src/c-writer.cc

```diff
@@ -1410,95 +1429,148 @@ void CWriter::WriteDataInitializers() {
             Write(Newline());
           Write(CloseBrace(), ";", Newline());
         }
-        ++data_segment_index;
       }
     }
-
-    memory = module_->memories[0];
   }
 
   Write("static void init_memory(", ModuleInstanceTypeName(), "* instance) ",
         OpenBrace());
   if (module_->memories.size() > module_->num_memory_imports) {
     Index memory_idx = module_->num_memory_imports;
     for (Index i = memory_idx; i < module_->memories.size(); i++) {
-      memory = module_->memories[i];
+      const Memory* memory = module_->memories[i];
       uint32_t max =
           memory->page_limits.has_max ? memory->page_limits.max : 65536;
       Write("wasm_rt_allocate_memory(", ExternalInstancePtr(memory->name), ", ",
             memory->page_limits.initial, ", ", max, ");", Newline());
     }
   }
-  data_segment_index = 0;
   for (const DataSegment* data_segment : module_->data_segments) {
-    Write("LOAD_DATA(", ExternalInstanceRef(memory->name), ", ");
-    WriteInitExpr(data_segment->offset);
-    Write(", data_segment_data_", data_segment_index, ", ",
-          data_segment->data.size());
-    Write(");", Newline());
-    ++data_segment_index;
+    if (data_segment->kind == SegmentKind::Active) {
```

Comment:
Below you do:

```
    if (elem_segment->kind != SegmentKind::Active) {
      continue;
    }
 ``` 

perhaps its worth being consistent?  (I don't feel strongly about which you should prefer).

Reply:
Done.